6.recursiveSum.js

```diff
@@ -1,17 +1,20 @@
 // Q6.
-// Implement a function that accepts an array of integers as an argument. 
-// This function should return the sum of each integer in the array. 
+// Implement a function that accepts an array of integers as an argument.
+// This function should return the sum of each integer in the array.
 // Your solution MUST be **recursive**. Your function must be pure (cannot use global variables)
-// 
+//
 // Example:
 // const arr1 = [5, 2, 9, 11]
 // recursiveSum(arr1) should return 27
 
-
 const recursiveSum = (arr) => {
+  // let i = 0
+  if (!arr.length) return 0;
+  // if (arr.length === 1) return arr[0];
+  // console.log(arr.pop())
+ 
+    return arr.pop() + recursiveSum(arr);
+  
+};
```

Comment:
This function wanted you to write it pure, so no side effects. Pop mutates your array so if you call the function two times in a row with the same array you'll get different results. The first result will be correct, the next will be 0 from the now empty array.